lib/message-handler.js

```diff
@@ -53,12 +53,50 @@ function messageHandlerFactory(
         // Speed up response time by queueing both these async actions
         return Promise.resolve()
         .then(function(){
-            return self.getDefaultBootfile(packetData);
+            /*
+             * Workaround for 'Intel Boot Agent' PXE boot issue that
+             * no free memory when PXE boot ROM donwload files for severail times.
+             * In some physical machine that use Intel Boot Agent PXE boot,
+             * when the boot sequence is:
+             *      NIC1 PXE boot
+             *   -> Donwload iPXE bootfile, and chainload to iPXE boot for NIC1
+             *   -> Noting for NIC1 iPXE boot, continue to NIC2 PXE boot
+             *   -> Noting for NIC2 PXE boot, continue to NIC1 PXE boot 
+             *   -> Donwload iPXE bootfile, and chainload to iPXE boot for NIC1
+             *   -> .....(loop as above)
+             * NIC ROM firmware will download iPXE bootfile in multiple times,
+             * but memory is not freed. So after several times, no free memory left.
+             * Console will show 'NBP is too big to fit in free base memory',
+             * which will lead to physical machine cannot auto-boot in next time.
+             *
+             * Workaround this issue by a pre-check. If it's not neccesary,
+             * don't need to send iPXE bootfile name to let node download bootfile.
+             *
+             * The pre-check could be configured so that when there's no this issue,
+             * it could be disabled, and also keep the ability of separate DHCP.
+             */
+            var dhcpSendBootFileName = configuration.get('dhcpSendBootFileName', true);
+            if(dhcpSendBootFileName) {
+                return self.isBootFileNameSent(packetData);
+            } else {
+                return true;
+            }
+        })
+        .then(function(isBootFileNameSent) {
+            if (isBootFileNameSent) {
```

Comment:
'bootfilename' is more accurate than 'bootfile', because dhcp only return bootfile name to pxe client, no bootfile is transferred.  and 'is' imply it's true/false value. so I think 'isBootFileNameSent' is more accurate.